src/pocketmine/entity/Living.php

```diff
@@ -786,10 +787,44 @@ public function getTargetBlock(int $maxDistance, array $transparent = []){
 			}
 		}catch(\ArrayOutOfBoundsException $e){
 		}
-
 		return null;
 	}
 
+        /**
+         *
+         * For the exact entity use a precision >= 0.998
+         *
+         * @param AxisAlignedBB|null $distance
+         * @param float              $precision
+         *
+         * @return null|Entity
+         *
+         */
+        public function getEntityLookingAt(AxisAlignedBB $distance = null, float $precision = 0.998){
+                if($distance === null){
+                        $pointA = $this->subtract(100, 100, 100);
+                        $pointB = $this->add(100, 100, 100);
+
+                        $distance = new AxisAlignedBB($pointA->x, $pointA->y, $pointA->z, $pointB->x, $pointB->y, $pointB->z);
```

Comment:
Following from my above comment, the first three parameters should be the lesser of each x,y,z and the second three should be the greater ones.